src/use-element-deletion-detection.hook.js

```diff
@@ -0,0 +1,20 @@
+import {useContext, useEffect} from 'react'
+import {RichTextContext} from './bandicoot.js'
+
+export function useElementDeletionDetection(domElement, cbk) {
+  const richTextContext = useContext(RichTextContext)
+
+  useEffect(() => {
+    if (domElement) {
+      richTextContext.addSelectionChangedListener(checkElementConnected)
+      return () => richTextContext.removeSelectionChangedListener(checkElementConnected)
+
+      function checkElementConnected() {
+        if (!domElement.isConnected && !domElement._bandicoot_delete_callback_called) {
```

Comment:
whenever the selection changes within the rich text editor, we check to see if the provided dom element is still "connected" to the dom. If it isn't, we call the callback and let the user of the hook know that the dom element has been deleted.